packages/gatsby-source-contentful/src/download-contentful-assets.js

```diff
@@ -9,7 +9,23 @@ const bar = new ProgressBar(
   }
 )
 
-let totalJobs = 0
+/**
+ * @name distributeWorkload
+ * @param workers A list of async functions to complete
+ * @param {number} count The number of task runners to use
+ */
+
+async function distributeWorkload(workers, count) {
+  const methods = workers.slice()
+
+  async function task() {
+    while (methods.length > 0) {
+      await methods.shift()
```

Comment:
While a change is necessary anyways; are you sure `.pop()` wouldn't be faster for really large arrays (200k+)? Internally it _ought_ to just move the offset pointer so I think it's equal.

Reply:
Good catch, just switched it. The code I adapted this from tried to keep some sense of order in the returned results so hence used shift.